usersession/agent/session_agent.go

```diff
@@ -201,6 +217,15 @@ func (s *SessionAgent) runServer() error {
 
 func (s *SessionAgent) shutdownServerOnKill() error {
 	<-s.tomb.Dying()
+	// closing the listener (but then it needs wrapping in
+	// closeOnceListener) before actually calling Shutdown, to
+	// workaround https://github.com/golang/go/issues/20239, we
+	// can in some cases (e.g. tests) end up calling Shutdown
+	// before runServer calls Serve and in go <1.11 this can be
+	// racy and the shutdown blocks.
```

Comment:
would it make sense to additionally check s.tomb.Alive() before calling Serve?

Reply:
that's still racy no? just more unlikely

Reply:
that's still racy, no? just more unlikely

Reply:
sorry, you said additionally...  what's your worry?